src/Components/Web.JS/src/Boot.Web.ts

```diff
@@ -50,6 +51,9 @@ function boot(options?: Partial<WebStartOptions>) : Promise<void> {
   const jsEventRegistry = JSEventRegistry.create(Blazor);
 
   const navigationEnhancementCallbacks: NavigationEnhancementCallbacks = {
+    enhancedNavigationStarted: (resource, options) => {
+      jsEventRegistry.dispatchEvent('enhancednavigationstart', { resource, options });
+    },
```

Comment:
This new event isn't strictly necessary - we could create internal hooks to enable `FocusOnNavigate` functionality.

However, it's pretty cheap to add, it helps with the implementation of `FocusOnNavigate`, and we've gotten some 👍s on #53887 from the community, indicating that this would be useful.

That said, I don't want to start a whole separate design discussion around this, especially since it's late in the release so we need to be wary about new features. So, if there's any objection to adding this here, I'm happy to remove it 🙂